src/main/java/com/parsing/parsers/pdf/parsing/RectangleDetector.java

```diff
@@ -211,26 +212,35 @@ public List<double[]> findVerticalLinesWithOpenCV(String fileSource) {
             double[] l = linesP.get(x, 0);
             lines.add(l);
         }
-        Imgcodecs.imwrite("verticalLines.png", linesP);
         return lines;
     }
 
     private List<double[]> findHorizontalLinesWithOpenCV(String fileSource) {
         Mat dst = new Mat();
         Mat cdst = new Mat();
+        Mat source;
         OpenCV.loadLocally();
-        Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
-        Imgproc.Canny(source, dst, ALL_LINES_THRESHOLD1, ALL_LINES_THRESHOLD2, ALL_LINES_APERTURESIZE, false);
-        Imgcodecs.imwrite("afterCannyRect.png", dst);
-        Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
-        Mat linesP = new Mat();
-        Imgproc.HoughLinesP(dst, linesP, ALL_LINES_RHO, Math.PI / 2, HORIZONTAL_THRESHOLD,
-                RectangleDetector.HORIZONTAL_LINE_LENGTH, ALL_LINES_MAXLINEGAP);
-        List<double[]> lines = new ArrayList<>();
-        for (int x = 0; x < linesP.rows(); x++) {
-            double[] l = linesP.get(x, 0);
-            lines.add(l);
+        if (fileSource != null) {
+            source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
+            Imgproc.Canny(source, dst, ALL_LINES_THRESHOLD1, ALL_LINES_THRESHOLD2, ALL_LINES_APERTURESIZE, false);
+            if (!dst.empty()) {
+                return new ArrayList<>();
+            } else {
+                Imgcodecs.imwrite("findhorizontalines_afterCannyRect.png", dst);
+                Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
+                Mat linesP = new Mat();
+                Imgproc.HoughLinesP(dst, linesP, ALL_LINES_RHO, Math.PI / 2, HORIZONTAL_THRESHOLD,
+                        RectangleDetector.HORIZONTAL_LINE_LENGTH, ALL_LINES_MAXLINEGAP);
+                List<double[]> lines = new ArrayList<>();
+                for (int x = 0; x < linesP.rows(); x++) {
+                    double[] l = linesP.get(x, 0);
+                    lines.add(l);
+                }
+                return lines;
+            }
+        } else {
+            log.debug("Filesource is null.");
```

Comment:
add a name of a file or something that will be useful

Reply:
Done.